build_runner/lib/src/build_script_generate/build_script_generate.dart

```diff
@@ -15,30 +15,135 @@ const scriptLocation = '.dart_tool/build/build.dart';
 Future<Null> ensureBuildScript() async {
   var scriptFile = new io.File(scriptLocation);
   // TODO - how can we invalidate this?
-  if (scriptFile.existsSync()) return;
+  //if (scriptFile.existsSync()) return;
   scriptFile.createSync(recursive: true);
   await scriptFile.writeAsString(await _generateBuildScript());
 }
 
+final _packageGraph = new TypeReference((b) => b
+  ..symbol = 'PackageGraph'
+  ..url = 'package:build_runner/build_runner.dart');
+
+final _buildAction = new TypeReference((b) => b
+  ..symbol = 'BuildAction'
+  ..url = 'package:build_runner/build_runner.dart');
+
+final _buildActions = new TypeReference((b) => b
```

Comment:
It is a bit confusing how there is also a method called `_buildActions`, maybe call this `_buildActionsType`?

Reply:
I split out the TypeReference definitions into a separate file imported with a prefix of `types`.